db/migrate/20230924161219_add_release_at_to_attachments.rb

```diff
@@ -0,0 +1,18 @@
+class AddReleaseAtToAttachments < ActiveRecord::Migration[6.0]
+  def up
+    add_column :attachments, :release_at, :datetime, default: -> { 'CURRENT_TIMESTAMP' }
+    Attachment.find_each do |attachment|
+      if attachment.released
+        attachment.update_attribute(:release_at, DateTime.now)
+      else
+        attachment.update_attribute(:release_at, DateTime.now + 1.year)
+      end
+    end
+  end
+  def down
+    Attachment.find_each do |attachment|
+      attachment.update_attribute(:released, attachment.release_at < DateTime.now)
+    end
+    remove_column :attachments, :release_at
+  end
+end
```

Comment:
update isn't the same as update_attribute, it takes a hash instead.

Not setting to nil because that would cause more special handling down the road. 1 year is reasonable.

Using released in the down migration should be fine since released should exist (if rolling back the migration after this one, which removed released)